File: euler/01/1.pl

use strict;
use warnings; 
use diagnostics;

my $sum = 0;
my $upper = $ARGV[0];
my $i;

if( not $upper ){
    $upper = 1000;
}

for(  $i = 0; $i<$upper; $i++){
    if( ($i % 3) == 0 || ($i % 5) == 0){
        $sum += $i;
    }
}

print "The total sum of multiples of 3 and 5 below $upper is $sum\n";

File: euler/01/1.cpp
#include <cstdio>
#include <cstdlib>

int main(int argc, char* argv[]){
	
	int upper;

	if(argc < 2){
		upper = 1000;
	}else{
		upper = atoi(argv[1]);
	}
	int sum = 0;

	int i=1;
	while(3*i < upper){
		sum += 3*i;
		++i;
	}
	
	i=1;
	while(5*(i+1)<upper){
		sum+= 5*(2*i+1);
		i+=3;
	}

	if(5*i<upper){
		sum += 5*i;
	}

	printf("The sum is %d\n", sum);
}
		


